core/src/client/mod.rs

```diff
@@ -734,4 +740,72 @@ impl StarknetClient for StarknetClientImpl {
 
         Ok(Address::from(evm_address_sliced))
     }
+
+    /// Returns token balances for a specific address given a list of contracts.
+    ///
+    /// # Arguments
+    ///
+    /// * `address(Address)` - specific address
+    /// * `contract_addresses(Vec<Address>)` - List of contract addresses
+    ///
+    /// # Returns
+    ///
+    ///  * `token_balances(TokenBalances)` - Token balances
+    ///
+    /// `Ok(<TokenBalances>)` if the operation was successful.
+    /// `Err(KakarotClientError)` if the operation failed.
+    async fn get_token_balances(
+        &self,
+        address: Address,
+        contract_addresses: Vec<Address>,
+    ) -> Result<TokenBalances, KakarotClientError> {
+        let mut token_balances = Vec::new();
+        let starknet_user_address = self
+            .compute_starknet_address(address, StarknetBlockId::Tag(BlockTag::Latest))
+            .await?;
+        let entrypoint = FieldElement::from_str("balanceOf").map_err(|e| {
+            KakarotClientError::OtherError(anyhow::anyhow!(
+                "Failed to convert entrypoint to FieldElement: {}",
+                e
+            ))
+        })?;
+
+        for token_address in contract_addresses.into_iter() {
+            let calldata = vec![entrypoint, starknet_user_address];
+            let call = self
+                .call_view(
+                    token_address,
+                    Bytes::from(vec_felt_to_bytes(calldata).0),
+                    StarknetBlockId::Tag(BlockTag::Latest),
+                )
+                .await;
+
+            let token_balance = match call {
+                Ok(call) => {
+                    let hex_balance = U256::from_str_radix(&call.to_string(), 16).map_err(|e| {
+                        KakarotClientError::OtherError(anyhow::anyhow!(
+                            "Failed to convert token balance to U256: {}",
+                            e
+                        ))
+                    })?;
+                    TokenBalance {
+                        contract_address: address,
+                        token_balance: Some(U256::from(hex_balance)),
+                        error: Some("".to_string()),
+                    }
+                }
+                Err(_) => TokenBalance {
+                    contract_address: address,
+                    token_balance: None,
+                    error: Some("Failed to get token balance".to_string()),
```

Comment:
just an idea, could be:
```
Err(e) => TokenBalance {
   contract_address: address,
   token_balance: None,
   error: Some(format!("kakarot_getTokenBalance Error:", e.to_string())
}

```

or something like that

could use also to_string_lossy or w/e works